sys/vm/swap_pager.c

```diff
@@ -2393,12 +2390,13 @@ swp_pager_meta_transfer(vm_object_t srcobject, vm_object_t dstobject,
 				swp_pager_meta_build(dstobject,
 				    sb->p + i - offset, sb->d[i], false);
 				VM_OBJECT_WLOCK(srcobject);
+			} else if (blk != SWAPBLK_NONE)
+				swp_pager_update_freerange(&range, sb->d[i]);
 #if __has_feature(capabilities)
+			if (blk != SWAPBLK_NONE)
 				swp_pager_cheri_xfer_tags(dstobject,
-				    sb->p + i - offset, sb, blk);
+				    sb->p + i - offset, sb, i);
```

Comment:
I should maybe commit this separately.  @markjdb noticed this bug which means we can copy the wrong tags